transactor/src/main/scala/io/mediachain/transactor/Copycat.scala

```diff
@@ -0,0 +1,186 @@
+package io.mediachain.transactor
+
+import scala.collection.mutable.{Set => MSet, HashSet => MHashSet, 
+                                 Map => MMap, HashMap => MHashMap}
+
+import io.atomix.copycat.{Command, Query}
+import io.atomix.copycat.server.{Commit, StateMachine, Snapshottable}
+import io.atomix.copycat.server.session.{ServerSession, SessionListener}
+import io.atomix.copycat.server.storage.snapshot.{SnapshotReader, SnapshotWriter}
+
+import cats.data.Xor
+
+object Copycat {
+  import io.mediachain.transactor.Types._
+  
+  case class JournalInsert(
+    record: CanonicalRecord
+  ) extends Command[Xor[JournalError, CanonicalEntry]]
+
+  case class JournalUpdate( 
+    ref: Reference,
+    cell: ChainCell
+  ) extends Command[Xor[JournalError, ChainEntry]]
+  
+  case class JournalLookup(
+    ref: Reference
+  ) extends Query[Option[Reference]]
+  
+  case class JournalCommitEvent(entry: JournalEntry)
+
+  class JournalStateMachine(
```

Comment:
oh you mean having the state machine on its own file instead of Copycat?
Probably a good idea -- I'll refactor once I have some more code to put in the different files.